From d7f55a0f6423e9b14861f607af19ccac4e7f31d5 Mon Sep 17 00:00:00 2001
Message-Id: <d7f55a0f6423e9b14861f607af19ccac4e7f31d5.1436312509.git.chang-joon.lee@intel.com>
From: "Kumar, Kiran S" <kiran.s.kumar@intel.com>
Date: Mon, 25 May 2015 14:12:42 +0530
Subject: [PATCH 01/15] REVERTME [VPG] drm/i915: Trigger flip on all active
 planes for z-order change

Currently when the z-order is changed flip is not forced on all active
planes and this manifest as flickers in some layers. To avoid this,
when ever the z-order changes, force flip on all active planes by
writing to the trigger register.

REVERTME: This patch fixes an issue in VPG internal set_display IOCTL.
This entire IOCTL and related fixes can be reverted once the upstream
nuclear page flip / atomic mode set lands in this product kernel.

Issue: IMINAN-35785
Change-Id: Iad721e40727a8fdd15b5711509cf96d39f8351d0
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 2c2369b..9a2e3f1 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11162,8 +11162,12 @@ int intel_set_disp_commit_regs(struct drm_mode_set_display *disp,
 	if (disp->update_flag & DRM_MODE_SET_DISPLAY_UPDATE_ZORDER) {
 		I915_WRITE_BITS(SPCNTR(intel_crtc->pipe, 0),
 				intel_crtc->reg.spacntr, 0x00000007);
+		I915_WRITE(SPSURF(intel_crtc->pipe, 0),
+				I915_READ(SPSURF(intel_crtc->pipe, 0)));
 		I915_WRITE_BITS(SPCNTR(intel_crtc->pipe, 1),
 				intel_crtc->reg.spbcntr, 0x00000007);
+		I915_WRITE(SPSURF(intel_crtc->pipe, 1),
+				I915_READ(SPSURF(intel_crtc->pipe, 1)));
 	}
 
 	/* Write to all display registers */
-- 
1.9.1

